roles/config-rh-sso/tasks/manage-roles.yml

```diff
@@ -17,14 +17,72 @@
   set_fact:
     role_json: "{{ lookup('template', 'realm_roles.json.j2') | to_json }}"
 
-- name: "Create Role for service Red Hat Single Sign-On"
-  uri:
-    url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/admin/realms/{{ role_data.targetRealm }}/roles"
-    method: POST
-    body: "{{ role_json }}"
-    body_format: json
-    status_code: 201, 409
-    headers:
-      Authorization: "Bearer {{ rh_sso_token.json.access_token }}"
-    validate_certs: "{{ rh_sso_ssl_cert_validation | default(omit) }}"
-  register: rh_sso_role_create
+- name: "Create Role: {{ role_data.name }} in Realm: {{ role_data.targetRealm }}"
+  block:
+    - name: "Create Role: {{ role_data.name }}"
+      uri:
+        url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/admin/realms/{{ role_data.targetRealm }}/roles"
+        method: POST
+        body: "{{ role_json }}"
+        body_format: json
+        status_code: 201, 409
+        headers:
+          Authorization: "Bearer {{ rh_sso_token.json.access_token }}"
+        validate_certs: "{{ rh_sso_ssl_cert_validation | default(omit) }}"
+      register: rh_sso_role_create
+    - name: "Update Role: {{ role_data.name }}"
+      uri:
+        url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/admin/realms/{{ role_data.targetRealm }}/roles/{{ role_data.name }}"
+        method: PUT
+        body: "{{ role_json }}"
+        body_format: json
+        status_code: 200, 204
+        headers:
+          Authorization: "Bearer {{ rh_sso_token.json.access_token }}"
+        validate_certs: "{{ rh_sso_ssl_cert_validation | default(omit) }}"
+      when:
+        - rh_sso_role_create.status == 409
+  when:
+    - role_data.clientRole is false
```

Comment:
Probably want to default this to avoid blowing up when not defined, e.g.:
```
- role_data.clientRole is undefined or role_data.clientRole is false
```